### DISCUSSION TITLE
[FEATURE] Add completions for the xonsh shell

### DISCUSSION THREAD

**Discussion by sbordeynemics** [fastapi/typer#875] [Questions]
### Is your feature request related to a problem

Is your feature request related to a problem? *Yes*

I've moved from ZSH to XONSH as my default shell, and I have multiple CLIs that rely on typer to work. Writing the completions by hand is doable, but it is time-consuming. Xonsh is a shell that mixes bash and python together, therefore completions are basically python functions.

### The solution you would like

cli --install-completions and cli --show-completions support for xonsh shells.

### Describe alternatives you've considered

I've considered hooking into my typer app to discover commands, arguments, options and so on to generate my completion automatically. It's doable, but at this point it should be a feature of the library, and not a hack.

**Comment by jgarte** [comment]
This would be awesome to have.

**Comment by svlandeg** [comment]
Hi! Thanks for the feature request. We're currently keeping track of these on the discussions forum, focusing the issue tracker mostly on bugs. I'll go ahead and move this thread but keep it open on the discussion forum so we can continue to follow up there.
